Makefile

```diff
@@ -0,0 +1,50 @@
+# Set Shell to bash, otherwise some targets fail with dash/zsh etc.
+SHELL := /bin/bash
+
+# Disable built-in rules
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-builtin-variables
+.SUFFIXES:
+.SECONDARY:
+.DEFAULT_GOAL := help
+
+include Makefile.vars.mk
+
+.PHONY: help
+help: ## Show this help
+	@grep -E -h '\s##\s' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: build
+build: build.bin build.docker ## All-in-one build
+
+.PHONY: build.bin
+build.bin: export CGO_ENABLED = 0
+build.bin: fmt vet ## Build binary
+	@go build -o $(BIN_FILENAME) main.go
+
+.PHONY: build.docker
+build.docker: build.bin ## Build docker image
+	docker build -t $(CONTAINER_IMG) .
+
+.PHONY: test
+test: test.go ## All-in-one test
+
+.PHONY: test.go
+test.go: ## Run unit tests against code
+	go test -race -coverprofile cover.out -covermode atomic -count 1 ./...
```

Comment:
Why do you force the retesting? (with `-count 1`) This can be helpful if the tests are non deterministic but I think for the vast majority of the applications I would prefer to keep test caching.

Reply:
I've come across a few cases in the past where the test cache was actually too aggressive and didn't execute test cases even after code changes, thus falsifying the results. Those were very hard to find out and debug. That's why I often "disable" that cache. However, maybe it was caused by an older Go version and fixed in the meantime :shrug: